src/core/mesh/qgsmeshdataprovidertemporalcapabilities.h

```diff
@@ -0,0 +1,139 @@
+#define QGSMESHDATAPROVIDERTEMPORALCAPABILITIES_H
+
+#include "qgsdataprovidertemporalcapabilities.h"
+#include "qgsrange.h"
+#include "qgsmeshdataset.h"
+
+
+/**
+ * \class QgsMeshDataProviderTemporalCapabilities
+ * \ingroup core
+ * Class for handling properties relating to a mesh data provider's temporal capabilities.
+ *
+ * \since QGIS 3.14
+ */
+class CORE_EXPORT QgsMeshDataProviderTemporalCapabilities: public QgsDataProviderTemporalCapabilities
+{
+  public:
+    /**
+     * Constructor for QgsMeshDataProviderTemporalCapabilities
+     */
+    QgsMeshDataProviderTemporalCapabilities();
+
+    /**
+     * Returns the last dataset index that has the time equal or lesser or than \a timeSinceGlobalReference (in milliseconds)from the dataset \a group
+     *
+     * Returns invalide dataset index if the time is less than the first dataset's time
+     */
+    QgsMeshDatasetIndex datasetIndexTimeInMilliseconds( int group, qint64 timeSinceGlobalReference ) const;
+
+    /**
+     * Returns the last dataset index that has the time equal or lesser or than \a timeSinceGlobalReference (in hours)from the dataset \a group
+     *
+     * Returns invalide dataset index if the time is less than the first dataset's time
+     */
+    QgsMeshDatasetIndex datasetIndex( int group, double  timeSinceGlobalReferenceInHours ) const;
+
+    /**
+     * Adds a \a reference date/time from a dataset \a group
+     *
+     * \note must be used only by the mesh data provider
+     */
+    void addGroupReferenceDateTime( int group, const QDateTime &reference ) SIP_SKIP;
+
+    /**
+     * Adds a \a time (in miliseonds) from a dataset contained in \a group
+     *
+     * \note must be used only by the mesh data provider,
+     * all dataset need to be added one after one
+     */
+    void addDatasetTimeInMilliseconds( int group, qint64 time ) SIP_SKIP;
+
+    /**
+     * Adds a \a time (in provider unit) from a dataset contained in \a group
+     *
+     * \note must be used only by the mesh data provider,
+     * all dataset need to be added one after one
+     */
+    void addDatasetTime( int group, double  time ) SIP_SKIP;
+
+    /**
+     * Returns true if the data has a reference time
+     *
+     */
+    bool hasReferenceTime() const;
+
+    /**
+     * Returns the reference time
+     *
+     */
+    QDateTime referenceTime() const;
+
+    /**
+     * Returns the time extent using the internal reference time
+     * and the first and last times available from the all the dataset
+     *
+     */
+    QgsDateTimeRange timeExtent() const;
+
+    /**
+     * Returns the time extent using \a reference date time
+     * and the first and last times available from the all the dataset
+     */
+    QgsDateTimeRange timeExtent( const QDateTime &reference ) const;
+
+    /**
+     * Sets the temporal unit (\a temporalUnit) used to read data
+     *
+     * Temporal units supported are milliseconds, seconds, minutes, hors, days and weeks
```

Comment:
do not state enum here, everyone can click on QgsUnitTypes::TemporalUnit to see options.

Reply:
hmmm, it is particular:
QgsUnitTypes::TemporalUnit siupport month, year, deca, ...
I have intentionally excluded time unit greater than week because there are not fixed duration time unit depending of the context (ho many day in a month, a year, ...). Those time units are not handle in the implemented conversion.
So, I have add doc about that.